resim/math/newton_solver.hh

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 ReSim, Inc.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include "resim/utils/nullable_reference.hh"
+#include "resim/utils/status_value.hh"
+
+namespace resim::math {
+
+template <int DIM>
+using DifferentiableFunction = std::function<Eigen::Matrix<double, DIM, 1>(
+    const Eigen::Matrix<double, DIM, 1> &,
+    NullableReference<Eigen::Matrix<double, DIM, DIM>>)>;
+
+template <int DIM>
```

Comment:
I think we should comment the interface here as well. (e.g. this does root-finding on fun, and tolerance is absolute tolerance)

Reply:
Good call! I've added some.